core/src/test/scala/org/apache/spark/sql/delta/ConvertToDeltaSQLSuite.scala

```diff
@@ -16,20 +16,47 @@
 
 package org.apache.spark.sql.delta
 
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
 import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.functions.{col, from_json}
 
 trait ConvertToDeltaSQLSuiteBase extends ConvertToDeltaSuiteBaseCommons
   with DeltaSQLCommandTest {
   override protected def convertToDelta(
       identifier: String,
-      partitionSchema: Option[String] = None): Unit = {
+      partitionSchema: Option[String] = None, collectStats: Boolean = true): Unit = {
     if (partitionSchema.isEmpty) {
-      sql(s"convert to delta $identifier")
+      sql(s"convert to delta $identifier ${collectStatisticsStringOption(collectStats)}")
     } else {
       val stringSchema = partitionSchema.get
-      sql(s"convert to delta $identifier partitioned by ($stringSchema) ")
+      sql(s"convert to delta $identifier ${collectStatisticsStringOption(collectStats)}" +
+        s" partitioned by ($stringSchema)")
     }
   }
+
+  // TODO: Move to ConvertToDeltaSuiteBaseCommons when DeltaTable API contains collectStats option
+  test("convert with collectStats set to false") {
+    withTempDir { dir =>
+      withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "true") {
+
+        val tempDir = dir.getCanonicalPath
+        writeFiles(tempDir, simpleDF)
+        convertToDelta(s"parquet.`$tempDir`", collectStats = false)
+        val deltaLog = DeltaLog.forTable(spark, tempDir)
+        val history = io.delta.tables.DeltaTable.forPath(tempDir).history()
+        checkAnswer(
+          spark.read.format("delta").load(tempDir),
+          simpleDF
+        )
+        assert(history.count == 1)
+        val statsDf = deltaLog.unsafeVolatileSnapshot.allFiles
+          .select(from_json(col("stats"), deltaLog.unsafeVolatileSnapshot.statsSchema)
+            .as("stats")).select("stats.*")
+        assert(statsDf.filter(col("numRecords").isNotNull).count == 0)
```

Comment:
we could add another assert that checks the sum of `numRecords` is equal to the number of records in `simpleDF`

Reply:
Done